rs/tests/ict/cmd/helpers.go

```diff
@@ -149,3 +151,101 @@ func get_closest_target_matches(all_targets []string, target string) []string {
 		return len(s) > 0
 	})
 }
+
+func sparse_checkout(repoUrl, repoDir string, sparseCheckoutPaths []string) (string, error) {
+	startingPoint, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("Could not get current dir: %v", err)
+	}
+
+	if repoDir == "" {
+		homedir, err := os.UserHomeDir()
+		if err != nil {
+			return "", fmt.Errorf("Failed to get home dir: %v", err)
+		}
+		repoDir = filepath.Join(homedir, ".cache", "k8s_repo")
+	}
+
+	if err := os.RemoveAll(repoDir); err != nil {
+		return return_to_starting_point(startingPoint, fmt.Errorf("Failed to remove directory: %v", err))
+	}
+
+	err = os.MkdirAll(repoDir, 0775)
+	if err != nil {
+		return return_to_starting_point(startingPoint, fmt.Errorf("Could not create repo directory: %v", err))
+	}
+
+	cloneCmd := exec.Command("git", "clone", "--filter=blob:none", "--no-checkout", repoUrl, repoDir)
+	stdErrBuffer := &bytes.Buffer{}
+	cloneCmd.Stderr = stdErrBuffer
+	if err := cloneCmd.Run(); err != nil {
+		return return_to_starting_point(startingPoint, fmt.Errorf("Failed to clone repository: %v\nStderr: %s", err, stdErrBuffer.String()))
+	}
+
+	if err := os.Chdir(repoDir); err != nil {
+		return return_to_starting_point(startingPoint, fmt.Errorf("Failed to chdir to repository: %v", err))
+	}
+
+	sparseCmd := exec.Command("git", "config", "core.sparseCheckout", "true")
```

Comment:
won't this modify the user's `git` config for that repo?

Reply:
From ict, i've made it use a different path (`~/.cache/k8s_repo`) which I configure. If ict is used, it will automatically try to sparse checkout the k8s repo. If you don't want to use the ict, then you can just give it the path by running bazel directly, for eg:
```bash
bazel test //rs/tests/testing_verification/testnets:from_config --config=systest --cache_test_results=no --test_env=IC_DASHBOARDS_DIR=/home/ubuntu/.cache/k8s_repo/bases/apps/ic-dashboards --sandbox_add_mount_pair=/home/ubuntu/.cache/k8s_repo/bases/apps/ic-dashboards --test_timeout=5400 --test_arg=--no-delete-farm-group --test_env=IC_CONFIG='{"subnets":[{"subnet_type":"system","num_nodes":4}],"unassigned_nodes":4}'
```
This wouldn't change the users repo config and it would just pass it through. Do you think we should support something like this in ict also? Feels unnecessary if we deem `ict` as a helper tool.